api/types/types.go

```diff
@@ -654,18 +654,19 @@ type Checkpoint struct {
 
 // Runtime describes an OCI runtime
 type Runtime struct {
-	Path string   `json:"path"`
+	Path string   `json:"path,omitempty"`
 	Args []string `json:"runtimeArgs,omitempty"`
 
-	// This is exposed here only for internal use
-	// It is not currently supported to specify custom shim configs
-	Shim *ShimConfig `json:"-"`
+	Shim *ShimConfig `json:"shim,omitempty"`
```

Comment:
From a UX perspective, the `--runtime` option was meant to be the more "convenient" option, where the user can use `docker info` to see which runtimes are available/configured on the daemon, and use those as option (`--runtime=runc`, `--runtime=kata`), and the daemon-configuratioin can contain additional config options (IIRC, this could even be multiple versions of `runc`, e.g.).


> No daemon.json configuration would be required to use alternative runtimes if the daemon was to interpret --runtime values

I know that access to the API already is equivalent to `root`, but if the `--runtime` value passed can be (as is being discussed here) either a name (`io.containerd.runc.v2`) or a path to a binary, that could potentially open up some "interesting" things (`--runtime=/bin/rm`, `--runtime=/some/arbitrary/script.sh`)